connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java

```diff
@@ -487,16 +489,20 @@ public void sendForm(FormWindow window, int id) {
         windowCache.showWindow(window, id);
     }
 
-    public void setRenderDistance(int renderDistance) {
-        renderDistance = GenericMath.ceil(++renderDistance * MathUtils.SQRT_OF_TWO); //square to circle
-        if (renderDistance > 32) renderDistance = 32; // <3 u ViaVersion but I don't like crashing clients x)
-        this.renderDistance = renderDistance;
+    public void setServerRenderDistance(int serverRenderDistance) {
+        serverRenderDistance = GenericMath.ceil(++serverRenderDistance * MathUtils.SQRT_OF_TWO); //square to circle
+        if (serverRenderDistance > 32) serverRenderDistance = 32; // <3 u ViaVersion but I don't like crashing clients x)
```

Comment:
We should probably get this squared away, or at least a better comment explaining when the crashing happens.

Reply:
Tested on ViaVersion 3.0.2 dev on Java 1.8 with absolutely no problems with a render distance of 64.